include/envoy/server/lifecycle_notifier.h

```diff
@@ -30,6 +30,15 @@ class ServerLifecycleNotifier {
     ShutdownExit
   };
 
+  // A handle to a registration that can be used to unregister.
+  class Handle {
+  public:
+    virtual ~Handle() = default;
+
+    virtual void unregister() PURE;
```

Comment:
drive by: Can we just have destruction of the Handle unregister. 

Bonus points: we have this pattern in quite a few different places, and I'm wondering if we could make this some type of generic mix-in class? WDYT?

Reply:
Moved unregister to the destructor.

Can you point me to some other examples of this pattern?